File: Examples/test-suite/java/native_directive_runme.java
import native_directive.*;

public class native_directive_runme {

  static {
    try {
	System.loadLibrary("native_directive");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {
    String s = "abc.DEF-123";
    if (native_directive.CountAlphas(s) != 6)
      throw new RuntimeException("CountAlphas failed");
    if (native_directive.CountAlphaCharacters(s) != 6)
      throw new RuntimeException("CountAlphaCharacters failed");
  }
}

File: Examples/test-suite/kotlin/native_directive_runme.kt
@file:JvmName("native_directive_runme")

import native_directive.*

fun main() {
    try {
        System.loadLibrary("native_directive")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    val s = "abc.DEF-123"
    if (native_directive.CountAlphas(s) != 6)
        throw RuntimeException("CountAlphas failed")
    // CountAlphaCharacters is provided by a Java-only %native directive (#if defined(SWIGJAVA))
    // and is therefore not generated for the Kotlin module.
}
